C/include/c4Listener.h

```diff
@@ -41,6 +41,25 @@ extern "C" {
     };
 
 
+    /** Called when a client connects, during the TLS handshake, if a client certificate is received.
+        @param listener  The C4Listener handling the connection.
+        @param clientCertData  The client's X.509 certificate in DER encoding.
+        @param context  The `tlsCallbackContext` from the `C4TLSConfig`.
+        @return  True to allow the connection, false to refuse it. */
+    typedef bool (*C4ListenerCertAuthCallback)(C4Listener *listener,
+                                               C4Slice clientCertData,
+                                               void *context);
+
+    /** Called when a client connects, after the TLS handshake (if any), when the initial HTTP request is
+        received.
+        @param listener  The C4Listener handling the connection.
+        @param authHeader  The "Authorization" header value from the client's HTTP request, or null.
+        @param context  The `callbackContext` from the listener config.
+        @return  True to allow the connection, false to refuse it. */
+    typedef bool (*C4ListenerHTTPAuthCallback)(C4Listener *listener,
+                                               C4Slice authHeader,
```

Comment:
Instead of passing the `authHeader` itself, can we decode and pass back the username and password? Since the fleece has the decoder and all platform has to do it.

Reply:
The advantage of this way is that later we can extend it later if something besides username and password is used in the auth header.  The auth header is not fleece, but base64 anyway.

Reply:
FYI, CBLHTTPLogic.m has code that shows how to generate & parse Authorization headers.